deploy/deploy.lib

```diff
@@ -317,8 +317,11 @@ configure_compilers() {
         log "...using gcc ${BASE_GCC_VERSION} from ${GCC_DIR}"
         if [[ ${spec} = *"intel"* ]]; then
             # update intel modules to use newer gcc in .cfg files
-            install_dir=$(spack location --install-dir ${sha})
-            for f in $(find ${install_dir} -name "icc.cfg" -o -name "icpc.cfg" -o -name "ifort.cfg"); do
+            INTEL_DIR=$(spack location --install-dir ${sha})
+            if [[ "${INTEL_DIR}" != "${SPACK_INSTALL_PREFIX}"* ]]; then
```

Comment:
See below; This part is run for *all* deployments, and the check keeps PR builds (which query the upstream install location) from modifying files in the real deployment. This has been more of an oversight that flew under the radar for the last two years. Probably no harm done since practically nothing changes for these code paths.